Source/ACE/Entity/Landblock.cs

```diff
@@ -157,8 +157,9 @@ public void AddWorldObject(WorldObject wo)
 
             lock (objectCacheLocker)
             {
-                allObjects = this.worldObjects.Values.ToList();
-                this.worldObjects[wo.Guid] = wo;
+                allObjects = worldObjects.Values.ToList();
+                if (!worldObjects.ContainsKey(wo.Guid))                                

```

Comment:
this wasn't at all necessary.  Dictionaries treat myDictionary[key] as an "add or set".